collect_app/src/main/java/org/odk/collect/android/preferences/ServerPreferencesFragment.java

```diff
@@ -354,21 +346,15 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
             case REQUEST_ACCOUNT_PICKER:
                 if (resultCode == RESULT_OK && data != null && data.getExtras() != null) {
                     String accountName = data.getStringExtra(AccountManager.KEY_ACCOUNT_NAME);
-                    if (accountName != null) {
-                        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getActivity());
-                        SharedPreferences.Editor editor = prefs.edit();
-                        editor.putString(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT, accountName);
-                        editor.apply();
-                        selectedGoogleAccountPreference.setSummary(accountName);
-                        credential.setSelectedAccountName(accountName);
-                    }
+                    accountsManager.setSelectedAccountName(accountName);
                 }
                 break;
         }
-        if (resultCode == RESULT_CANCELED) {
-            ToastUtils.showShortToast("Account selection cancelled");
-        }
-
     }
 
+    @Override
+    public void googleAccountSelected() {
+        String accountName = accountsManager.getCredentials().getSelectedAccountName();
```

Comment:
Instead of reaching into the accountsManager, we could just pass the String via this call: `googleAccountSelected(String accountName)` as we have access to it at the call site.